pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/snyk/go-application-framework/pkg/local_workflows/content_type"
+	"github.com/snyk/go-application-framework/pkg/local_workflows/local_models"
+	"github.com/snyk/go-application-framework/pkg/utils"
+	"github.com/snyk/go-application-framework/pkg/workflow"
+	"github.com/spf13/pflag"
+)
+
+const (
+	FilterFindingsWorkflowName = "findings.filter"
+)
+
+var WORKFLOWID_FILTER_FINDINGS = workflow.NewWorkflowIdentifier(FilterFindingsWorkflowName)
+
+func InitFilterFindingsWorkflow(engine workflow.Engine) error {
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func filterFindingsBySeverity(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return utils.Contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func filterFindings(findingsModel local_models.LocalFinding, filter []FindingsFilterFunc) local_models.LocalFinding {
+	filteredFindings := local_models.LocalFinding{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filter {
+			if match {
+				match = filter(finding)
+			}
+		}
+		if match {
+			filteredFindings.Findings = append(filteredFindings.Findings, finding)
+		}
+	}
+	return filteredFindings
+}
+
+func filterFindingsEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	ff_transform_enabled := config.GetBool(configuration.FF_TRANSFORMATION_WORKFLOW)
+	output = []workflow.Data{}
+
+	logger.Println("Workflow Name:", FilterFindingsWorkflowName)
+
+	if !ff_transform_enabled {
+		return input, nil
+	}
+
+	var findingsModels []local_models.LocalFinding
+
+	for _, data := range input {
+		if strings.HasPrefix(data.GetContentType(), content_type.LOCAL_FINDING_MODEL) {
+			var findingsModel local_models.LocalFinding
+			findingsBytes, ok := data.GetPayload().([]byte)
+			if !ok {
+				var findingsError snyk_errors.Error = snyk_errors.Error{
+					Title:          "Invalid Payload Type",
+					Classification: "Internal",
+					Level:          "warning",
+					Detail:         "Invalid payload type",
+				}
+				logger.Err(findingsError).Msg(findingsError.Error())
+				return nil, findingsError
```

Comment:
Question: Do we really want to stop the data flow if this workflow can't process all the data? I know this goes in the direction of panic vs not. But I feel there is a bit of uncertainty here as the error we create says "warning", but effectively we return nil so we swallow all data. And the message "Invalid Payload Type" is not helping to resolve the issue easily if it at all happens.